rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -27,168 +57,154 @@ func Qualify_AssetClass_InterestRate: <"Qualifies a product as having the Asset
-                    then (Qualify_UnderlierProduct_Equity(optionUnderlier) = True
-                            or optionUnderlier -> index ->> assetClass = AssetClassEnum -> Equity
-                            or optionUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> Equity)
-                    else False)
-            or (economicTerms -> payout -> forwardPayout only exists
-                and if forwardUnderlier exists
-                    then (Qualify_UnderlierProduct_Equity(forwardUnderlier)
-                            or forwardUnderlier -> index ->> assetClass = AssetClassEnum -> Equity
-                            or forwardUnderlier -> security -> productTaxonomy -> primaryAssetClass any = AssetClassEnum -> Equity)
-                    else False)
+                and (ObservableQualification(optionUnderlier -> Observable, SecurityTypeEnum -> Equity, AssetClassEnum -> Equity) 
+                    or if optionUnderlier -> Product exists
+                        then Qualify_AssetClass_Equity(optionUnderlier -> Product ->> economicTerms) = True
+                        else False
+                ))
+            or (economicTerms -> payout -> settlementPayout only exists
+                and (UnderlierQualification(
+						settlementUnderlier, empty, AssetClassEnum -> Equity) 
+                    or if settlementUnderlier -> Product ->> economicTerms exists
+                        then Qualify_AssetClass_Equity(
+									settlementUnderlier -> Product ->> economicTerms) = True
+                        else False
+                ))
 
 func Qualify_AssetClass_Commodity: <"Qualifies a product as having the Asset Class classification Commodity">
     inputs:
         economicTerms EconomicTerms (1..1)
     output:
         is_product boolean (1..1)
 
-    alias optionUnderlier: economicTerms -> payout -> optionPayout only-element -> underlier
+    alias optionUnderlier: 
+        economicTerms -> payout -> optionPayout only-element -> optionUnderlier
 
-    alias forwardUnderlier:
-        economicTerms -> payout -> forwardPayout only-element -> underlier
+    alias settlementUnderlier:
+        economicTerms -> payout -> settlementPayout only-element -> underlier
 
     set is_product:
         // CO:FixFlo
         ((economicTerms -> payout -> commodityPayout, economicTerms -> payout -> fixedPricePayout) only exists
-                and economicTerms -> payout -> commodityPayout -> underlier -> commodity exists)
+                and economicTerms -> payout -> commodityPayout -> underlier exists)
                     // CO Basis
             or (economicTerms -> payout -> commodityPayout only exists
                 and economicTerms -> payout -> commodityPayout count = 2
-                and economicTerms -> payout -> commodityPayout -> underlier -> commodity count = 2)
+                and economicTerms -> payout -> commodityPayout -> underlier count = 2)
```

Comment:
Stop path at `commodityPayout` and count there.

Reply:
done